### PR TITLE
Report metric command queue size

### PR DESCRIPTION
**Pull request by HaoranYi**
#### Problem

https://github.com/solana-labs/solana/issues/24319

When logger is spammed, system will use more memory. However, there is no clear indicator of how many pending metrics are in the queue.

#### Summary of Changes

Add a report of metric command queue size.

Fixes #

### CODE REVIEW

File: metrics/src/metrics.rs [hunk 1]
```diff
@@ -208,6 +209,7 @@ impl MetricsAgent {
                 .add_field_i64("points_written", points_written as i64)
                 .add_field_i64("num_points", num_points as i64)
                 .add_field_i64("points_lost", (num_points - points_written) as i64)
+                .add_field_i64("pending_command_queue_size", cmd_queue_size as i64)
```
Comment (behzadnouri) [inline metrics/src/metrics.rs:212]: I would rather name this more consistently with other fields here; such as `points_buffered`.

`command_queue` is an implementation detail, and wouldn't necessarily make sense to someone inspecting this metric on the dashboards or metrics UI.

File: metrics/src/metrics.rs [hunk 2]
```diff
@@ -186,6 +186,7 @@ impl MetricsAgent {
         max_points: usize,
         max_points_per_sec: usize,
         last_write_time: Instant,
+        cmd_queue_size: usize,
```
Comment (behzadnouri) [inline metrics/src/metrics.rs:189]: maybe rename to `points_buffered` for naming consistency with other args.

**Review by behzadnouri — APPROVED** [review]

File: metrics/src/metrics.rs (hunk 1 above)
Comment (HaoranYi) [reply metrics/src/metrics.rs]: updated.

**Review by behzadnouri — APPROVED** [review]

**Review by behzadnouri — APPROVED** [review]
lgtm

**Comment by HaoranYi** [comment]
image
One rpc node running for the past 30 minutes. The max qlen is 568 so far.

**Pull request merged**